### PR TITLE
reformat version string (remove markdown format)

### LINKED ISSUE #5136: Widelands Version String Generates Erratic Markdown
Development versions of Widelands have the format
```
versionnumber ~ revision [ commit @ branch ] ( buildtype )
```
e.g.
```
1.1~git25529[fe7992d@master] (Release)
```
Unfortunately, when posting something containing `[hello](world)` in a Markdown-interpreted form (e.g. a GitHub issue or the forum turns this into a link to `world` displayed as `hello`. This makes the pasted version string much harder to read, and on GitHub it additionally prevents GitHub from automatically linkifying the commit hash (as in: fe7992d). This does not happen if there's a whitespace between `]` and `(`, but the version string is inconsistently displayed sometimes with and sometimes without this whitespace.

I propose to change all occurrences of this version string formatting to use different characters that are never interpreted by Markdown even if some whitespace is added or removed, for example by replacing the `()` with `[]` or vice versa or using `{}/$%+#`. While doing this it should also be made consistent everywhere.

Technically, only the part `1.1~git25529` is the actual version number (used for version comparison in packages and releases); everything after this is just extra formatting for display which we can modify freely.

### PR DESCRIPTION
**Pull request by andy5995**
(fixes #5136)

I thought #5330 was merged and I deleted the branch, which closed the PR. So I'm reopening this now. I apologize for the inconvenience, if any.

### CODE REVIEW

File: src/editor/editorinteractive.cc [hunk 1]
```diff
@@ -991,9 +991,9 @@ void EditorInteractive::run_editor(UI::Panel* error_message_parent,
 		   format(
 		      _("An error has occured. The error message is:\n\n%1$s\n\nPlease report "
 		        "this problem to help us improve Widelands. You will find related messages in the "
-		        "standard output (stdout.txt on Windows). You are using build %2$s "
-		        "(%3$s).\nPlease add this information to your report."),
-		      e.what(), build_id(), build_type()),
+		        "standard output (stdout.txt on Windows). You are using build %2$s.\n"
```
Comment (Noordfrees) [inline src/editor/editorinteractive.cc]:
```suggestion
		        "standard output (stdout.txt on Windows). You are using version %2$s.\n"
```

File: src/main.cc [hunk 2]
```diff
@@ -36,8 +36,7 @@
  * Cross-platform entry point for SDL applications.
  */
 int main(int argc, char* argv[]) {
-	std::cout << "This is Widelands Version " << build_id() << " (" << build_type() << ")"
-	          << std::endl;
+	std::cout << "This is Widelands Version " << build_ver_details() << std::endl;
```
Comment (Noordfrees) [inline src/main.cc]:
```suggestion
	std::cout << "This is Widelands version " << build_ver_details() << std::endl;
```

File: src/wlapplication.cc [hunk 3]
```diff
@@ -1603,9 +1603,9 @@ void WLApplication::emergency_save(UI::Panel* panel,
 	if (ask_for_bug_report) {
 		log_err("  Please report this problem to help us improve Widelands.\n"
 		        "  You will find related messages in the standard output (stdout.txt on Windows).\n"
-		        "  You are using build %s (%s).\n"
+		        "  You are using build %s.\n"
```
Comment (Noordfrees) [inline src/wlapplication.cc]:
```suggestion
		        "  You are using version %s.\n"
```

File: src/wlapplication.cc [hunk 4]
```diff
@@ -1619,9 +1619,9 @@ void WLApplication::emergency_save(UI::Panel* panel,
 		   format(
 		      _("An error has occured. The error message is:\n\n%1$s\n\nPlease report "
 		        "this problem to help us improve Widelands. You will find related messages in the "
-		        "standard output (stdout.txt on Windows). You are using build %2$s "
-		        "(%3$s).\nPlease add this information to your report."),
-		      error, build_id(), build_type()),
+		        "standard output (stdout.txt on Windows). You are using build %2$s.\n"
```
Comment (Noordfrees) [inline src/wlapplication.cc]:
```suggestion
		        "standard output (stdout.txt on Windows). You are using version %2$s.\n"
```

File: src/wlapplication.cc [hunk 5]
```diff
@@ -1635,12 +1635,12 @@ void WLApplication::emergency_save(UI::Panel* panel,
             format(
 		         _("An error occured during the game. The error message is:\n\n%1$s\n\nPlease report "
 		           "this problem to help us improve Widelands. You will find related messages in the "
-		           "standard output (stdout.txt on Windows). You are using build %2$s "
-		           "(%3$s).\n\nPlease add this information to your report.\n\nWould you like "
+		           "standard output (stdout.txt on Windows). You are using build %2$s.\n\n"
```
Comment (Noordfrees) [inline src/wlapplication.cc]:
```suggestion
		           "standard output (stdout.txt on Windows). You are using version %2$s.\n\n"
```

File: src/wlapplication_messages.cc [hunk 6]
```diff
@@ -245,15 +245,13 @@ bool is_parameter(const std::string& name) {
...
-                const std::string& build_type,
-                CmdLineVerbosity verbosity) {
+void show_usage(const std::string& build_ver_details, CmdLineVerbosity verbosity) {
 	i18n::Textdomain textdomain("widelands_console");
 
 	std::cout << std::string(kIndent + kTextWidth, '=')
 	          << std::endl
 	          /** TRANSLATORS: %s = version information */
-	          << format(_("This is Widelands Version %s"), format("%s(%s)", build_id, build_type))
+	          << format(_("This is Widelands Version %s"), format("%s", build_ver_details))
```
Comment (Noordfrees) [inline src/wlapplication_messages.cc]:
```suggestion
	          << format(_("This is Widelands version %s"), build_ver_details)
```

File: utils/detect_revision.py [hunk 7]
```diff
@@ -65,8 +65,8 @@ def detect_git_revision():
             ['git', 'rev-parse', '--abbrev-ref', 'HEAD'], cwd=base_path)
         git_abbrev = stdout.rstrip()
         if git_count and git_revnum and git_abbrev:
-            # 1.0~git34567[abc0123@master]
-            return '%s~git%s[%s@%s]' % (next_stable_version, git_count, git_revnum, git_abbrev)
+            # 1.0~git34567 (abc0123@master)
+            return '%s~git%s (%s@%s)' % (next_stable_version, git_count, git_revnum, git_abbrev)
```
Comment (Noordfrees) [inline utils/detect_revision.py:69]: You'll also need to adapt the return value for the other functions (`detect_bzr_revision` in lines 105 and 120, and maybe others)

**Review by Noordfrees — COMMENTED** [review]
Got around to trying this out now, a few comments inline for consistency

**Comment by andy5995** [comment]
rebased at https://github.com/widelands/widelands/commit/3c65ca457c7b1e62f5241baa6bdf3e6f4c8d3ec6

File: src/base/log.cc [hunk 8]
```diff
@@ -82,8 +82,7 @@ class WindowsLogger {
 		std::cout.rdbuf(stdout_.rdbuf());
 		std::cerr.rdbuf(stderr_.rdbuf());
 		// Repeat version info so that we'll have it available in the log file too
-		std::cout << "This is Widelands Version " << build_id() << " (" << build_type() << ")"
-		          << std::endl;
+		std::cout << "This is Widelands Version " << build_ver_details() << std::endl;
```
Comment (matthiakl) [inline src/base/log.cc]: One more in the "consistency" category:
```suggestion
		std::cout << "This is Widelands version " << build_ver_details() << std::endl;
```
The rest looks good now. I don't find any other occurrences of the version.

**Review by matthiakl — APPROVED** [review]
LGTM now.

**Pull request merged**